test/tools/ossfuzz/lpsolver/FuzzerSolverInterface.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <test/tools/ossfuzz/lpsolver/FuzzerSolverInterface.h>
+
+#include <range/v3/view/enumerate.hpp>
+
+using namespace solidity::test::fuzzer::lpsolver;
+using namespace solidity::util;
+using namespace std;
+using namespace z3;
+
+FuzzerSolverInterface::FuzzerSolverInterface(bool _supportModels):
+	m_lpSolver(_supportModels),
+	m_z3Solver(m_z3Ctx)
+{
+	m_lpSolvingState.variableNames.emplace_back("");
+}
+
+LinearExpression FuzzerSolverInterface::linearExpression(vector<int> _factors)
+{
+	LinearExpression lexp;
+	lexp.resize(_factors.size());
+	for (auto&& [index, value]: _factors | ranges::views::enumerate)
+		lexp[index] = rational{value};
+	return lexp;
+}
+
+void FuzzerSolverInterface::addLPConstraint(pair<bool, vector<int>> _constraint)
+{
+	m_lpSolvingState.constraints.push_back({linearExpression(move(_constraint.second)), _constraint.first});
+}
+
+void FuzzerSolverInterface::addZ3Constraint(
+	pair<bool, vector<int>> _constraint,
+	vector<expr>& _symbolicVars
+)
+{
+	bool isEquality = _constraint.first;
+	expr reduce(m_z3Ctx);
+	unsigned count = 0;
+	for (auto&& [index, value]: _constraint.second | ranges::views::enumerate)
+	{
+		if (index != 0)
+			if (value != 0)
+				reduce = reduce + expr(value * _symbolicVars[count++]);
```

Comment:
shouldn't this be
```suggestion
				reduce = reduce + expr(value * _symbolicVars[index]);
```
?